api/test/unit/services/application.service.spec.ts

```diff
@@ -768,6 +773,108 @@ describe('Testing application service', () => {
     });
   });
 
+  it('should get public apps view information when applications are available', async () => {
+    const userId = randomUUID();
+    const requestingUser = {
+      id: userId,
+      firstName: 'requesting fName',
+      lastName: 'requesting lName',
+      email: 'requestingUser@email.com',
+      jurisdictions: [{ id: 'juris id' }],
+    } as unknown as User;
+    const date = new Date();
+    const mockedValues = mockApplicationSet(3, date);
+    const listingStatuses = [
+      { status: ListingsStatusEnum.active },
+      { status: ListingsStatusEnum.closed },
+      {
+        status: ListingsStatusEnum.closed,
+        lotteryStatus: LotteryStatusEnum.publishedToPublic,
+      },
+    ];
+    const mockedValuesWithListing = mockedValues.map((mockedValue, idx) => {
+      return {
+        ...mockedValue,
+        listings: listingStatuses[idx],
+      };
+    });
+    prisma.applications.findMany = jest.fn().mockResolvedValue(
```

Comment:
Is it worth mocking this value to this detail if I am only checking the length? I considered matching with more detail as well but felt that would be more complexity than value since the function isn'y transforming the response? Thoughts?

Reply:
Also any tips to reduce the duplicate mocking code here?

Reply:
You can define all the static-ish variables, like the userid, requesting user, mocked values, statuses, and even the response under the describe block and shared them. If you want to overwrite specific parts of it, you can just spread ie { ...mainMock, customField: 'binky' }

Reply:
Just cleaned things up with some shared variables! Thanks!